spec/integration/parser/parameter_defaults_spec.rb

```diff
@@ -26,12 +26,16 @@
 
     let (:compiler) { Puppet::Parser::Compiler.new(Puppet::Node.new('specification')) }
 
+    let (:topscope) { compiler.topscope }
+
     def collect_notices(code)
       logs = []
       Puppet[:code] = code
       Puppet::Util::Log.with_destination(Puppet::Test::LogCollector.new(logs)) do
-        compiler.compile
-        yield
+        Puppet.override(:global_scope => topscope) do
+          compiler.compile
+          yield
+        end
```

Comment:
I think  a more accurate fix here would be:
```
compiler.compile do |catalog|
  yield
  catalog
end
```
The yield will then end up within the override performed by the compiler and no extra override is necessary.